src/MagicalClass.java

```diff
@@ -0,0 +1,89 @@
+import java.util.ArrayList;
+
+public abstract class MagicalClass extends PlayableCharacter {
+
+ 	// Attributes
+	private final int maxMana = 100;
+	private int mana;
+	private int inteligence;
+	protected ArrayList<Spell> spellList;
+	protected static final String errorMessage = "Error: Invalid value";
+
+	//Constructors
+	public MagicalClass(String name, int mana, int inteligence) throws Exception {
+		super(name);
+		this.setMana(mana);
+		this.setInteligence(inteligence);
+	}
+
+ 	public MagicalClass(byte level, String name, int experience, int mana, int inteligence) throws Exception {
+		super(level,name,experience);
+		this.setMana(mana);
+		this.setInteligence(inteligence);
+	}
+ 	
+ 	// Methods
+ 	public abstract void lernNewSpell(Spell newSpell);
+
+	public void forgetSkill(int id) {
+		Spell toRemove = null;
+		for (Spell spell : this.getSpellList()) {
+			if(spell.getId() == id) {
+				toRemove = spell;
+			}
+		}
+		if(toRemove != null) {
+			this.spellList.remove(toRemove);
+		}
+	}
+ 	
```

Comment:
I don't remember correctly, but I think seba told us something about checking if the list was empty before cycling throw it. And maybe we could use Exception here too? what do you think?

Reply:
Yup! I will fix it